src/rabbit_fifo.erl

```diff
@@ -974,26 +976,48 @@ append_to_master_index(RaftIdx,
     State#?MODULE{ra_indexes = Indexes}.
 
 
+incr_enqueue_count(#?MODULE{cfg = #cfg{release_cursor_interval = C} = Cfg}
+                   = State0)
+  when is_integer(C) ->
+    %% conversion to new release cursor interval format
+    State = State0#?MODULE{cfg = Cfg#cfg{release_cursor_interval = {C, C}}},
+    incr_enqueue_count(State);
```

Comment:
This function clause could go after the next one:
* the most common clause appears first, which could be more readable for developpers
* this is a small optimization (the most common clause is evaluated first)

Reply:
Likewise for other function clauses which do the conversion.

Reply:
Hmm yes that may actually apply to pretty much all "overloaded" functions. I'll update.

Reply:
That said the compiler might do this optimisation for us.